File: tests/degree-of-success.test.ts
import {DegreeOfSuccess, determineDegreeOfSuccess} from '../src/degree-of-success';

describe('degree of success', () => {
    test('normal dc tests', () => {
        expect(determineDegreeOfSuccess(10, 10, 20)).toBe(DegreeOfSuccess.CRITICAL_FAILURE);
        expect(determineDegreeOfSuccess(10, 10, 11)).toBe(DegreeOfSuccess.FAILURE);
        expect(determineDegreeOfSuccess(10, 10, 10)).toBe(DegreeOfSuccess.SUCCESS);
        expect(determineDegreeOfSuccess(10, 10, 0)).toBe(DegreeOfSuccess.CRITICAL_SUCCESS);
    });

    test('rolling 1 dc tests', () => {
        expect(determineDegreeOfSuccess(1, 10, 20)).toBe(DegreeOfSuccess.CRITICAL_FAILURE);
        expect(determineDegreeOfSuccess(1, 10, 11)).toBe(DegreeOfSuccess.CRITICAL_FAILURE);
        expect(determineDegreeOfSuccess(1, 10, 10)).toBe(DegreeOfSuccess.FAILURE);
        expect(determineDegreeOfSuccess(1, 10, 0)).toBe(DegreeOfSuccess.SUCCESS);
    });

    test('rolling 20 dc tests', () => {
        expect(determineDegreeOfSuccess(20, 10, 20)).toBe(DegreeOfSuccess.FAILURE);
        expect(determineDegreeOfSuccess(20, 10, 11)).toBe(DegreeOfSuccess.SUCCESS);
        expect(determineDegreeOfSuccess(20, 10, 10)).toBe(DegreeOfSuccess.CRITICAL_SUCCESS);
        expect(determineDegreeOfSuccess(20, 10, 0)).toBe(DegreeOfSuccess.CRITICAL_SUCCESS);
    });
});

File: src/commonTest/kotlin/at/posselt/pfrpg2e/data/checks/DegreeOfSuccessTest.kt
package at.posselt.pfrpg2e.data.checks

import at.posselt.pfrpg2e.data.checks.DegreeOfSuccess.*
import kotlin.test.Test
import kotlin.test.assertEquals

class DegreeOfSuccessTest {
    @Test
    fun degreeOfSuccess() {
        assertEquals(CRITICAL_FAILURE, determineDegreeOfSuccess(10, 0, dieValue = 2))
        assertEquals(CRITICAL_FAILURE, determineDegreeOfSuccess(10, 9, dieValue = 1))
        assertEquals(FAILURE, determineDegreeOfSuccess(10, 0, dieValue = 20))
        assertEquals(FAILURE, determineDegreeOfSuccess(10, 9, dieValue = 2))
        assertEquals(FAILURE, determineDegreeOfSuccess(10, 10, dieValue = 1))
        assertEquals(SUCCESS, determineDegreeOfSuccess(10, 10, dieValue = 2))
        assertEquals(SUCCESS, determineDegreeOfSuccess(10, 20, dieValue = 1))
        assertEquals(CRITICAL_SUCCESS, determineDegreeOfSuccess(10, 10, dieValue = 20))
        assertEquals(CRITICAL_SUCCESS, determineDegreeOfSuccess(10, 20, dieValue = 2))
    }
}